docs/src/main/sphinx/optimizer/cost-based-optimizations.rst

```diff
@@ -10,54 +10,53 @@ Join enumeration
 The order in which joins are executed in a query can have a significant impact
 on the query's performance. The aspect of join ordering that has the largest
 impact on performance is the size of the data being processed and transferred
-over the network. If a join, that produces a lot of data, is performed early in
-the execution, then subsequent stages need to process large amounts of
-data for longer than necessary, increasing the time and resources needed for
-the query.
+over the network. If a join which produces a lot of data is performed early in
+the query's execution, then subsequent stages need to process large amounts of
+data for longer than necessary, increasing the time and resources needed for the
+query.
 
-With cost based join enumeration, Trino uses
-:doc:`/optimizer/statistics` provided by connectors to estimate
-the costs for different join orders and automatically picks the
-join order with the lowest computed costs.
+With cost-based join enumeration, Trino uses :doc:`/optimizer/statistics`
+provided by connectors to estimate the costs for different join orders and
+automatically picks the join order with the lowest computed costs.
 
 The join enumeration strategy is governed by the ``join_reordering_strategy``
 session property, with the ``optimizer.join-reordering-strategy``
```

Comment:
This is the first place a session property is referenced in the doc, I would suggest making "session property" a :ref: link to `session-properties-definition` which goes here: https://trino.io/docs/current/sql/set-session.html#session-properties